src/components/sidebar/SidebarNavigation/index.tsx

```diff
@@ -9,30 +10,32 @@ import {
   SidebarListItemText,
 } from '@/components/sidebar/SidebarList'
 import { navItems } from './config'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { AppRoutes } from '@/config/routes'
 
 const Navigation = (): ReactElement => {
   const router = useRouter()
-  const query = { safe: router.query.safe }
-  const [openHref, setOpenHref] = useState<string>(router.pathname)
+  const { safe } = useSafeInfo()
 
-  useEffect(() => {
-    setOpenHref(router.pathname)
-  }, [router.pathname, query.safe])
+  const needsUpdate = safe.implementationVersionState === ImplementationVersionState.OUTDATED
 
   return (
     <SidebarList>
-      {navItems.map((item) => (
-        <ListItem key={item.href} disablePadding selected={router.pathname === item.href}>
-          <SidebarListItemButton
-            onClick={() => setOpenHref(item.href)}
-            selected={openHref === item.href}
-            href={{ pathname: item.href, query }}
-          >
-            {item.icon && <SidebarListItemIcon>{item.icon}</SidebarListItemIcon>}
-            <SidebarListItemText bold>{item.label}</SidebarListItemText>
-          </SidebarListItemButton>
-        </ListItem>
-      ))}
+      {navItems.map((item) => {
+        const isSettings = item.href === AppRoutes.settings.setup
+
+        return (
+          <ListItem key={item.href} disablePadding selected={router.pathname === item.href}>
+            <SidebarListItemButton
+              selected={router.pathname === item.href}
+              href={{ pathname: item.href, query: { safe: router.query.safe } }}
+            >
+              {item.icon && <SidebarListItemIcon badge={isSettings && needsUpdate}>{item.icon}</SidebarListItemIcon>}
+              <SidebarListItemText bold>{item.label}</SidebarListItemText>
+            </SidebarListItemButton>
+          </ListItem>
+        )
+      })}
```

Comment:
I would try and avoid conditions inside a loop.
E.g.

```
// Indicate that the Safe needs an upgrade
if (safe.implementationVersionState === ImplementationVersionState.OUTDATED) {
  const setupItem = navItems.find((item) => item.href === AppRoutes.settings.setup)
  setupItem?.badge = true
}
```
The navItem type needs to support the `badge` prop, obviously.

Then just map over items like before and pass the badge prop transparently.